Upgrade search services and requirements

yvanzo opened the pull request:
## Problem

After having changed the MusicBrainz database schema version to 30 in the pull request #305 and releasing it as [v-2025-05-20.0-mbdb30](https://github.com/metabrainz/musicbrainz-docker/releases/tag/v-2025-05-20.0-mbdb30), this patch additionally does:

* [MBVM-100] Upgrade the required version of Docker Compose to 2
* [MBVM-101] Upgrade MusicBrainz Solr major version to 4 (Apache Solr 9)
  (which has been committed earlier then reverted for the previous release)
* Upgrade SIR major version to 4 (Python 3)
* Upgrade MusicBrainz search schema (mbsssss) to [v-2025-05-18](https://github.com/metabrainz/mbsssss/releases/tag/v-2025-05-18).

## Todolist

* [x] Upgrade SIR version to 4.0.1 (to be released)
* [x] Make up-to-date Solr 9 collection backups available
* [x] Test in an up-to-date mirror instance

yvanzo committed:
[MBVM-101] Upgrade MB Solr version to 4.0.0 (#304)

This patch upgrades to MB Solr version to 4,
which is mainly an upgrade to the upstream Apache Solr 9.

Querying the Solr API doesn’t change much, at least not enough to
require any changes to the indexer and the musicbrainz server.

However, there is a breaking change in the way the data is stored.
It requires either re-indexing all the data, or restoring the data
from the newly available SolrCloud 8.9+ collections backup archives.
The old Solr 7 data will be automatically removed at the first run.

Main changes are:

- MusicBrainz XML Metadata (MMD) Schema version is v-2025-04-23

- Solr major version is 9
  (upgrading from Solr 7)

- Solr runs in SolrCloud mode
  (that is the future standard and allows for loading backups)

- Swappiness is reduced for optimization

- Volume is mounted to /var/solr and includes logs too

- The default command is a custom script that:
  - installs solr.xml configuration file
  - removes Solr 7 data if present
  - creates empty SolrCloud 9 collections

- Helper commands to fetch, load and clean up Solr backup archives are
  now included within the Docker image/container of the search service,
  moving from the `admin/` directory in the Docker Compose host.

  A new helper command is also available in the same Docker
  image/container to delete all the indexed documents if needed.

  The script to fetch the Postgres database has been updated
  to just check the day of the latest Solr backup instead.

Drawbacks are:

- The size of the new backup archives is ~20% larger than the previous
  search index dumps, currently ~60 GB rather than ~50 GB, to fetch.

- Calling the new backup restore API requires the extracted backup
  files, in parallel of the Solr data repository being populated.
  That takes much more disk space: ~80G more.

yvanzo committed:
Upgrade MB Solr version to 4.1.0

For users, it brings support MBIDs for artist credits and medium.

For deployment, the default command has been set.

It brings a different search schema and requires reindexing cdstub and release.

See release notes for detailed upgrade steps.

Reference: https://github.com/metabrainz/mb-solr/releases/tag/v4.1.0

yvanzo committed:
Upgrade SIR version to 4.0.0

SIR has been ported to Python 3 and uses the MB DB schema 30 - 2025 Q2.

It is indexing more details and requires reindexing most collections.

See release notes for detailed upgrade steps.

Reference: https://github.com/metabrainz/sir/releases/tag/v4.0.0

yvanzo committed:
Drop Travis CI which is no longer used since 2021

yvanzo committed:
[MBVM-100] Drop support for Docker Compose v1

- Docker Compose v1 was a separate script named `docker-compose`.
  Docker v2 comes with `compose` as a subcommand to `docker`.

- The top-level `version` property is obsolete in Docker v2.

References:
* https://docs.docker.com/compose/releases/migrate/
* https://docs.docker.com/reference/compose-file/version-and-name/#version-top-level-element-obsolete
* https://blog.metabrainz.org/2024/09/27/major-upgrade-for-mirrors-november-25-2024/

yvanzo committed:
Assume non-root user in 'docker' group on host

Most users have a `docker` group and don’t need the `sudo` command.

Keep supporting both use cases but favor the most common and simple one.

Reference: https://docs.docker.com/engine/install/linux-postinstall/

yvanzo commented:
Rebased on the current `HEAD` of the branch `master` tagged `v-2025-05-20.0-mbdb30`.

yvanzo committed:
Update SIR to the bug-fixing version 4.0.1

Reference: https://github.com/metabrainz/sir/releases/tag/v4.0.1

yvanzo commented:
Updated to SIR 4.0.1 and fix docs.

yvanzo commented:
Applied amending commits and rebased on the current `HEAD` of the branch `master` tagged `v-2025-05-23.0-mbdb30`.

yvanzo commented:
When running:

```sh
docker compose create search
```

got the message:

    Your kernel does not support memory swappiness capabilities or the cgroup is not mounted. Memory swappiness discarded.

which refers to adding `mem_swappiness: 1`:

https://github.com/metabrainz/musicbrainz-docker/pull/306/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R95

Reference: https://solr.apache.org/guide/solr/9_7/deployment-guide/taking-solr-to-production.html#avoid-swapping-nix-operating-systems

yvanzo commented:
I tested both installing a mirror from scratch, and updating a mirror from Solr 7 to Solr 9.
In the later case, you also want to remove the previous `searchdump` volume after the update:

```bash
docker volume rm musicbrainz-docker_searchdump
```

(The volume name has been changed to avoid file permission issues when using the new helper scripts.)

mwiencek reviewed (approved):
Thanks, I just reviewed each of the commits, it looks ready to go.

yvanzo committed:
Drop misleading check broken after search upgrades

This check has always been misleading because the search index is
asynchronous, hence the counts cannot match but on a stopped replica.

Since the script is broken due to the use of 4 shards for recordings
with SolrCloud 9, and isn't relevant, it seems wiser to just drop it.

yvanzo committed:
Document the remaining helper scripts

yvanzo commented:
Further update/document helper scripts under `admin/` with the commits https://github.com/metabrainz/musicbrainz-docker/pull/306/commits/64021a5cb0b7a685dcc70bb7105b098174d63d78 and https://github.com/metabrainz/musicbrainz-docker/pull/306/commits/ec00233633f56a53623d17d3e86e23674047eee4.

yvanzo commented:
Last documentation updates.
